File: src/opcbiz/fxprimus/dto/register/RegisterDto.py
from collections import namedtuple
from src.opcbiz.fxprimus.dto import ExcelFieldDto


class RegisterDto:
    def __init__(self, id, title, account_type, first_name, last_name, email, country, phone_number, password,
                 checkbox_agreement, checkbox_kid, checkbox_marketing, checkbox_understand):
        self.id = id
        self.title = title

        self.account_type = account_type
        self.first_name = first_name
        self.last_name = last_name
        self.email = email
        self.country = country
        self.phone_number = phone_number
        self.password = password

        self.checkbox_agreement = checkbox_agreement
        self.checkbox_kid = checkbox_kid
        self.checkbox_marketing = checkbox_marketing
        self.checkbox_understand = checkbox_understand

    def custom_dto_decoder(self):
        return namedtuple('RegisterDto', self.keys())(*self.values())

    @staticmethod
    def set_email(self, email):
        self.email = email

    @staticmethod
    def excel_template():
        excel_field_dto_list = [ExcelFieldDto.ExcelFieldDto('ID', 'id'),
                                ExcelFieldDto.ExcelFieldDto('TITLE', 'title'),
                                ExcelFieldDto.ExcelFieldDto('ACCOUNT_TYPE', 'account_type'),
                                ExcelFieldDto.ExcelFieldDto('FIRST_NAME', 'first_name'),
                                ExcelFieldDto.ExcelFieldDto('LAST_NAME', 'last_name'),
                                ExcelFieldDto.ExcelFieldDto('EMAIL', 'email'),
                                ExcelFieldDto.ExcelFieldDto('COUNTRY', 'country'),
                                ExcelFieldDto.ExcelFieldDto('PHONE_NUMBER', 'phone_number'),
                                ExcelFieldDto.ExcelFieldDto('PASSWORD', 'password'),

                                ExcelFieldDto.ExcelFieldDto('CHECKBOX_AGREEMENT', 'checkbox_agreement'),
                                ExcelFieldDto.ExcelFieldDto('CHECKBOX_KID', 'checkbox_kid'),
                                ExcelFieldDto.ExcelFieldDto('CHECKBOX_MARKETING', 'checkbox_marketing'),
                                ExcelFieldDto.ExcelFieldDto('CHECKBOX_UNDERSTAND', 'checkbox_understand')]
        return excel_field_dto_list

File: FXPRIMUS_JAVA/src/test/java/com/opcbiz/fxprimus/dto/pendCompleteRegis/RegisterDto.java
package com.opcbiz.fxprimus.dto.pendCompleteRegis;

import java.util.ArrayList;
import java.util.List;

import com.opcbiz.fxprimus.dto.ExcelFieldDto;

public class RegisterDto {
	String id;
	String title;
	String url;
	String accountType;
	String firstName;
	String lastName;
	String email;
	String country;
	String phoneNumber;
	String password;
	String displayCheckboxValidation;
	String typeAccount;
	String buttonRegisterValidation;
	String profileOptionValidation;
	String dashboardValidation;

	

	public static List<ExcelFieldDto> excelTemplate() {
		List<ExcelFieldDto> excelFieldDtoList = new ArrayList<ExcelFieldDto>();

		excelFieldDtoList.add(new ExcelFieldDto("ID", "id", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("TITLE", "title", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("URL", "url", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList
				.add(new ExcelFieldDto("ACCOUNT_TYPE", "accountType", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("FIRST_NAME", "firstName", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("LAST_NAME", "lastName", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("EMAIL", "email", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("COUNTRY", "country", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList
				.add(new ExcelFieldDto("PHONE_NUMBER", "phoneNumber", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("PASSWORD", "password", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("DISPLAY_CHECKBOX_VALIDATION", "displayCheckboxValidation",
				ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList
				.add(new ExcelFieldDto("TYPE_ACCOUNT", "typeAccount", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("BUTTON_REGISTER_VALIDATION", "buttonRegisterValidation",
				ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("PROFILE_OPTION_VALIDATION", "profileOptionValidation",
				ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("DASHBOARD_VALIDATION", "dashboardValidation",
				ExcelFieldDto.DataType.STRING, "STRING"));
		return excelFieldDtoList;
	}
	public RegisterDto(String id, String title, String url, String accountType, String firstName, String lastName,
			String email, String country, String phoneNumber, String password, String displayCheckboxValidation,
			String typeAccount, String buttonRegisterValidation, String profileOptionValidation,
			String dashboardValidation) {
		super();
		this.id = id;
		this.title = title;
		this.url = url;
		this.accountType = accountType;
		this.firstName = firstName;
		this.lastName = lastName;
		this.email = email;
		this.country = country;
		this.phoneNumber = phoneNumber;
		this.password = password;
		this.displayCheckboxValidation = displayCheckboxValidation;
		this.typeAccount = typeAccount;
		this.buttonRegisterValidation = buttonRegisterValidation;
		this.profileOptionValidation = profileOptionValidation;
		this.dashboardValidation = dashboardValidation;
	}
	public RegisterDto() {
		super();
	}
	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public String getAccountType() {
		return accountType;
	}

	public void setAccountType(String accountType) {
		this.accountType = accountType;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getCountry() {
		return country;
	}

	public void setCountry(String country) {
		this.country = country;
	}

	public String getPhoneNumber() {
		return phoneNumber;
	}

	public void setPhoneNumber(String phoneNumber) {
		this.phoneNumber = phoneNumber;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getDisplayCheckboxValidation() {
		return displayCheckboxValidation;
	}

	public void setDisplayCheckboxValidation(String displayCheckboxValidation) {
		this.displayCheckboxValidation = displayCheckboxValidation;
	}

	public String getTypeAccount() {
		return typeAccount;
	}

	public void setTypeAccount(String typeAccount) {
		this.typeAccount = typeAccount;
	}

	public String getButtonRegisterValidation() {
		return buttonRegisterValidation;
	}

	public void setButtonRegisterValidation(String buttonRegisterValidation) {
		this.buttonRegisterValidation = buttonRegisterValidation;
	}

	public String getProfileOptionValidation() {
		return profileOptionValidation;
	}

	public void setProfileOptionValidation(String profileOptionValidation) {
		this.profileOptionValidation = profileOptionValidation;
	}

	public String getDashboardValidation() {
		return dashboardValidation;
	}

	public void setDashboardValidation(String dashboardValidation) {
		this.dashboardValidation = dashboardValidation;
	}
}
